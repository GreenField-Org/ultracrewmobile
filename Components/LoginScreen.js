import React, { useState } from 'react';
import { StyleSheet, Text, TextInput, View, Image, TouchableOpacity } from 'react-native';
import { useFonts } from 'expo-font';

const customFont = {
  'Lobster-Regular': require('../assets/fonts/Lobster-Regular.ttf')
}

export default function LoginScreen() {
  //it's binnnaa while since I've used state. But I think this is right
  const [username, setUsername] = useState('')
  const [password, setPassword] = useState('')
  let [fontsLoaded] = useFonts(customFont)
  
  if (fontsLoaded) {
    return (
      <View style={styles.container}>
        <Text style={styles.titleText}>Ultra Crew App</Text>
        <View style={styles.inputView}>
        
          <TextInput
            style={styles.TextInput}
            placeholder="email"
            placeholderTextColor="#003f5c"
            onChangeText={(email) => setUsername(username)}
          />
        </View>
 
        <View style={styles.inputView}>
          <TextInput
            style={styles.TextInput}
            placeholder="password"
            placeholderTextColor="#003f5c"
            //should we add option to show pw when you're done typing it?
            secureTextEntry={true}
            onChangeText={(password) => setPassword(password)}
          />
        </View>

        <TouchableOpacity style={styles.loginButton}>
          <Text style={styles.loginText}>Login</Text>
        </TouchableOpacity>

        <TouchableOpacity>
          <Text style={styles.createAcct}
            //add onpress.. use react dom to link to new page?
          >Create an account</Text>
        </TouchableOpacity>
      </View>
    )
  } else {
    return (
      <View />
    )
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  titleText: {
    fontSize: 40,
    marginBottom: 50,
    fontFamily: 'Lobster-Regular'
  },
  inputView: {
    backgroundColor: "lightgray",
    borderRadius: 5,
    width: "80%",
    height: 45,
    marginBottom: 20,
    alignItems: "center",
  },
  TextInput: {
    height: 50,
    flex: 1,
    fontSize: 20,
    padding: 5,
    marginLeft: 20,
  },

  createAcct: {
    marginTop: 40
  },

  loginButton:
 {
   width:"80%",
   borderRadius:10,
   height:50,
   fontSize: 20,
   alignItems:"center",
   justifyContent:"center",
   backgroundColor:"orange",
 }

});
