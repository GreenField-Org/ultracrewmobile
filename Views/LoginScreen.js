import React, { useState } from 'react';
import { StyleSheet, Text, TextInput, View, Image, TouchableOpacity } from 'react-native';
import { useFonts } from 'expo-font';
import DefaultButton from '../Components/DefaultButton';

export default function LoginScreen() {
  //it's binnnaa while since I've used state. But I think this is right
  const [username, setUsername] = useState('')
  const [password, setPassword] = useState('')
  let [fontsLoaded] = useFonts({
    'Lobster-Regular': require('../assets/fonts/Lobster-Regular.ttf')
  })
  
  return (
    <View style={styles.container}>
    <Text style={styles.titleText}> Ultra Crew App</Text>
      <View style={styles.inputView}>
  
  <TextInput
    style={styles.TextInput}
    placeholder="email"
    placeholderTextColor="#003f5c"
    onChangeText={(email) => setUsername(username)}
    
  />
</View>
 
<View style={styles.inputView}>
  <TextInput
    style={styles.TextInput}
    placeholder="password"
    placeholderTextColor="#003f5c"
    //should we add option to show pw when you're done typing it?
    secureTextEntry={true}
    onChangeText={(password) => setPassword(password)}
  />
</View>

<DefaultButton buttonText='Login'/>

<TouchableOpacity style={styles.loginButton}>
<Text>Create an Account</Text>
</TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  createAcctBtn:{
    width: 50
  },
  container: {
    flex: 1,
    backgroundColor: '#F8CD82',
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 150
  },

  titleText: {
    fontSize: 40,
    marginBottom: 40,
    fontFamily: 'Lobster-Regular'
  },

  image :{
    marginBottom: 40,
    height: 100, 
    width: 200,
    borderRadius: 10
  },

  inputView: {
    backgroundColor: "lightgray",
    borderRadius: 5,
    width: "80%",
    height: 45,
    marginBottom: 20,
    alignItems: "center",
  },
  
  TextInput: {
    height: 50,
    flex: 1,
    fontSize: 20,
    padding: 5,
    marginLeft: 20,
  },

  createAcct: {
    marginTop: 40
  },

  loginButton:
 {
   width:"80%",
   borderRadius:10,
   height:50,
   fontSize: 20,
   alignItems:"center",
   justifyContent:"center",
   backgroundColor:"#0DF7DB",
 }

});
