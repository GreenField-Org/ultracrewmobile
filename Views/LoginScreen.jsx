import React, { useState } from "react";
import { Pressable } from "react-native";
import { StyleSheet, Text, TextInput, View } from "react-native";
import { useFonts } from "expo-font";

const customFont = {
  "Lobster-Regular": require("../assets/fonts/Lobster-Regular.ttf"),
};

export default function LoginScreen({ navigation }) {
    const [username, setUsername] = useState("");
    const [password, setPassword] = useState("");
    const [fontsLoaded] = useFonts(customFont);

    if (fontsLoaded) {
        return (
          <View style={styles.container}>
            <Text style={styles.titleText}>Ultra Crew App</Text>
            <View style={styles.inputView}>
              <TextInput
                style={styles.TextInput}
                placeholder="email"
                placeholderTextColor="#003f5c"
                onChangeText={(email) => setUsername(username)}
              />
            </View>

            <View style={styles.inputView}>
              <TextInput
                style={styles.TextInput}
                placeholder="password"
                placeholderTextColor="#003f5c"
                //should we add option to show pw when you're done typing it?
                secureTextEntry={true}
                onChangeText={(password) => setPassword(password)}
              />
            </View>

            <Pressable
              title="Login"
              style={styles.loginButton}
              onPress={() => navigation.navigate("Home")}
            >
              <Text style={styles.text}>Login</Text>
            </Pressable>

            <Pressable
              title="Create an Account"
              style={styles.createAcctBtn}
              onPress={() => navigation.navigate("CreateAcct")}
            >
              <Text style={styles.createAcctBtn}>Create an Account</Text>
            </Pressable>
          </View>
        )
    } else {
        return <View />
    }
}

const styles = StyleSheet.create({
    createAcctBtn: {
        fontSize: 20,
    },
    container: {
        backgroundColor: "#F8CD82",
        alignItems: "center",
        justifyContent: "center",
        flex: 1,
    },
    titleText: {
        fontSize: 50,
        marginBottom: 50,
        fontFamily: "Lobster-Regular",
    },
    inputView: {
        backgroundColor: "#EBEBEB",
        borderRadius: 10,
        width: "80%",
        height: 45,
        marginBottom: 20,
        alignItems: "center",
    },
    TextInput: {
        height: 50,
        flex: 1,
        fontSize: 20,
        paddingTop: 5,
        paddingBottom: 5,
        marginLeft: 20,
        textAlign: "center",
    },
    createAcct: {
        marginTop: 60,
    },
    loginButton: {
        width: "80%",
        borderRadius: 10,
        marginBottom: 35,
        height: 50,
        fontSize: 30,
        alignItems: "center",
        justifyContent: "center",
        backgroundColor: "#0DF7DB",
        fontWeight: 700,
    },
});
