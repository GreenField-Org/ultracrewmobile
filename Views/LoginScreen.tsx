import React, { useState } from "react";
import { Pressable } from "react-native";
import { StyleSheet, Text, TextInput, View } from "react-native";
import { useFonts } from "expo-font";

const customFont = {
    "Lobster-Regular": require("../assets/fonts/Lobster-Regular.ttf"),
};

export default function LoginScreen({ navigation }) {
    const [username, setUsername] = useState("");
    const [password, setPassword] = useState("");
    const [fontsLoaded] = useFonts(customFont);

    if (fontsLoaded) {
        return (
            <View style={styles.container}>
                <Text style={styles.titleText}>Ultra Crew App</Text>
                <View style={styles.loginBox}>
                    <TextInput
                        style={styles.TextInput}
                        placeholder="email"
                        placeholderTextColor="#003f5c"
                        onChangeText={(email) => setUsername(username)}
                        inputMode="email"
                    />
                    <TextInput
                        style={styles.TextInput}
                        placeholder="password"
                        placeholderTextColor="#003f5c"
                        //should we add option to show pw when you're done typing it?
                        secureTextEntry={true}
                        onChangeText={(password) => setPassword(password)}
                    />

                    <Pressable
                        style={styles.loginButton}
                        onPress={() => navigation.navigate("Home")}
                    >
                        <Text style={styles.text}>Login</Text>
                    </Pressable>

                    <Pressable
                        onPress={() => navigation.navigate("CreateAcct")}
                    >
                        <Text style={styles.createAcctBtn}>
                            Create an Account
                        </Text>
                    </Pressable>
                </View>
            </View>
        );
    } else {
        return <View />;
    }
}

const styles = StyleSheet.create({
    createAcctBtn: {
        fontSize: 14,
        color: "#FF9001",
    },
    container: {
        backgroundColor: "#1c1c1c",
        alignItems: "center",
        justifyContent: "center",
        flex: 1,
    },
    titleText: {
        fontSize: 50,
        marginBottom: 50,
        paddingTop: 10,
        fontFamily: "Lobster-Regular",
        color: "#FF9001",
    },
    loginBox: {
        alignItems: "center",
        borderColor: "#ebebeb",
        borderWidth: 1,
        boxShadow: "#FF9001",
        borderTopLeftRadius: 30,
        borderTopRightRadius: 30,
        width: "100%",
        paddingVertical: 100,
        paddingHorizontal: 10,
        marginTop: 50,
        bottom: -30,
    },
    text: {
        fontSize: 20,
        fontWeight: "600",
    },
    TextInput: {
        backgroundColor: "#C4C4C4",
        height: 60,
        width: "80%",
        borderRadius: 50,
        flex: 1,
        fontSize: 20,
        marginVertical: 10,
        padding: 10,
        textAlign: "center",
    },
    createAcct: {
        marginTop: 60,
        marginBottom: 60,
    },
    loginButton: {
        width: "40%",
        borderRadius: 50,
        marginBottom: 35,
        marginTop: 10,
        height: 50,
        alignItems: "center",
        justifyContent: "center",
        backgroundColor: "#0DF7DB",
        fontWeight: "700",
    },
});
