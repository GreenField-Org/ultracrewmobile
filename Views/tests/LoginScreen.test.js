import LoginScreen from "../LoginScreen";
import renderer, {act} from 'react-test-renderer'


describe('Login Screen', () =>{
    it('', ()=>{
        expect(renderer.create(<LoginScreen/>)).toMatchSnapshot()
    })
})