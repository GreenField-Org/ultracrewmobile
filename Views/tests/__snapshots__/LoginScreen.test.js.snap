// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen  1`] = `<View />`;
